bluemira/equilibria/fem_fixed_boundary/transport_solver.py

```diff
@@ -0,0 +1,186 @@
+
+"""
+The bluemira transport solver module
+"""
+from abc import ABC, abstractmethod
+
+import numpy as np
+from scipy.interpolate import interp1d
+
+from bluemira.codes.plasmod.api import RunMode, Solver
+from bluemira.codes.plasmod.mapping import Profiles
+
+
+class TransportSolver(ABC):
+    """Abstract transport solver class"""
+
+    # This abstract class specifies all the properties and methods that
+    # should be provided by a generic transport solver. For the moment
+    # they are limited to pprime and ffprime. It is only a draft version.
+    # TODO: see #1448
+
+    def __init__(self, *args, **kwargs):
+        self.solver = None
+
+    @property
+    @abstractmethod
+    def pprime(self):
+        """
+        Get pprime as function of psi_norm
+        """
+        pass
+
+    @property
+    @abstractmethod
+    def ffprime(self):
+        """
+        Get ffprime as function of psi_norm
+        """
+        pass
+
+
+class NoneTransportSolver(TransportSolver):
+    """Empty transport solver"""
+
+    def __init__(self, *args, **kwargs):
+        self.solver = None
+
+    @TransportSolver.pprime.getter
+    def pprime(self):
+        """
+        Get pprime as function of psi_norm
+        """
+        return None
+
+    @TransportSolver.ffprime.getter
+    def ffprime(self):
+        """
+        Get ffprime as function of psi_norm
+        """
+        return None
+
+
+class PlasmodTransportSolver(TransportSolver):
+    """
+    PLASMOD transport solver class
+    """
+
+    def __init__(self, params, build_config):
+        self.solver = Solver(params, build_config)
+        self._x = None
+        self._pprime = None
+        self._ffprime = None
+        self._psi = None
+        self._kappa_95 = None
+        self._delta_95 = None
+        self._volume_in = None
+        self._I_p = None
+
+    def execute(self):
+        """
+        Run the PLASMOD transport solver
+        """
+        self.solver.execute(RunMode.RUN)
+        # supporting variables
+
+        # the normalization on x_phi is made because the x profile given by plasmod is
+        # not well normalized and it is given in a range slightly different than [0,1].
+
+        self.__x_phi = self.solver.get_profile(Profiles.x)
+        self.__x_phi /= np.max(self.__x_phi)
+        self.__psi_plasmod = self.solver.get_profile(Profiles.psi)
+        self.__x_psi = np.sqrt(self.__psi_plasmod / self.__psi_plasmod[-1])
+
+    def _from_phi_to_psi(self, profile):
+        """
+        Convert the profile to the magnetic coordinate sqrt((psi - psi_ax)/(psi_b -
+        psi_ax))
+        """
+        profile_data = self.solver.get_profile(profile)
+        interp_data = interp1d(self.__x_psi, profile_data, kind="linear")
+        return interp_data(self.__x_phi)
+
+    @property
+    def x(self):
+        """Get the magnetic coordinate"""
+        # This variable is "reduntant" since it is equale to the renormalized x_phi.
```

Comment:
```suggestion
        # This variable is "redundant" since it is equal to the renormalized x_phi.
```

Reply:
Is this comment worth putting in the docstring? I can't see another way to publicly access `x_phi` though, so this doesn't appear to be redundant? Or am I missing something?